src/components/spin-button/spin-button.component.ts

```diff
@@ -93,4 +95,77 @@ export class SpinButtonComponent implements ControlValueAccessor {
         this.disabled = isDisabled;
     }
 
+    onKeypress(event: KeyboardEvent): boolean {
+
+        // we only need to perform checks if the type is number
+        if (this.type !== 'number') {
+            return;
+        }
+
+        if (!this._regex.test(event.key)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    onPaste(event: ClipboardEvent): void {
+
+        // we only need to perform checks if the type is number
+        if (this.type !== 'number') {
+            return;
+        }
+
+        // get the value being pasted
+        const value = event.clipboardData.getData('text');
+
+        // check if it contains the character
+        if (!this._regex.test(value)) {
+
+            // inset the numeric value only if there is one
+            const numericValue = parseFloat(value);
+
+            if (!isNaN(numericValue)) {
+                this.value = numericValue;
+            }
+
+            event.stopPropagation();
+            event.preventDefault();
+        }
+    }
+
+    onValueChange(input: HTMLInputElement, value: string | number): void {
+
+        // convert any numeric value to a string
+        let valueAsString = typeof value === 'number' ? value.toString() : value;
+
+        if (valueAsString.length < this.maxLength && this.type === 'number') {
+            valueAsString = '0' + valueAsString;
```

Comment:
This causes problems when `maxLength` is unset (i.e. Infinity), as seen at https://plnkr.co/edit/BzKEf820MuSF6hKmfoRZ?p=preview